plugins/inputs/consul/consul.go

```diff
@@ -111,6 +123,22 @@ func (c *Consul) GatherHealthCheck(acc telegraf.Accumulator, checks []*api.Healt
 		tags["service_name"] = check.ServiceName
 		tags["check_id"] = check.CheckID
 
+		if c.GatherAllTags {
+			for _, checkTag := range check.ServiceTags {
+				if c.TagDelimiter != "" && len(c.TagDelimiter) == 1 {
+					r, _ := regexp.Compile("\\w+" + c.TagDelimiter + "\\w+")
```

Comment:
I don't think we need a regex, just `SplitN(tag, delimiter, 2)`, and then assign the correct tag value based on the len of the returned value.